### ISSUE TITLE
Arrays: callback builder

### ISSUE THREAD

**Issue by ljacqu** [ljacqu/JavaTypeResolver#98]
Depends on #97:

If I want to delegate to different methods based on array type, it would be nice to have a builder. Though the benefit is not huge since any reasonable IDE will generate a `switch` statement over the enum type, and there's some "fighting" involved with `void` methods
